modular_pariah/modules/contractor/code/datums/contractor_items.dm

```diff
@@ -136,74 +149,44 @@
 	desc = "Request Syndicate Command to distrupt the station's powernet. Disables power across the station for a short duration."
 	item_icon = "bolt"
 	limited = 2
-	cost = 3
+	cost = 2
 
 /datum/contractor_item/blackout/handle_purchase(datum/contractor_hub/hub)
 	. = ..()
 
-	if (.)
-		power_fail(35, 50)
-		priority_announce("Abnormal activity detected in [station_name()]'s powernet. As a precautionary measure, the station's power will be shut off for an indeterminate duration.", "Critical Power Failure", ANNOUNCER_POWEROFF)
-
-// Subtract cost, and spawn if it's an item.
-/datum/contractor_item/proc/handle_purchase(datum/contractor_hub/hub, mob/living/user)
-
-	if (hub.contract_rep >= cost)
-		hub.contract_rep -= cost
-	else
-		return FALSE
-
-	if (limited >= 1)
-		limited -= 1
-	else if (limited == 0)
-		return FALSE
-
-	hub.purchased_items.Add(src)
-
-	user.playsound_local(user, 'sound/machines/uplinkpurchase.ogg', 100)
-
-	if (item && ispath(item))
-		var/atom/item_to_create = new item(get_turf(user))
-
-		if(user.put_in_hands(item_to_create))
-			to_chat(user, span_notice("Your purchase materializes into your hands!"))
-		else
-			to_chat(user, span_notice("Your purchase materializes onto the floor."))
-
-		return item_to_create
-	return TRUE
+	if(!(.))
+		return
+	power_fail(35, 50)
+	priority_announce("Abnormal activity detected in [station_name()]'s powernet. As a precautionary measure, the station's power will be shut off for an indeterminate duration.", "Critical Power Failure", ANNOUNCER_POWEROFF)
 
-/obj/item/pinpointer/crew/contractor
-	name = "contractor pinpointer"
-	desc = "A handheld tracking device that locks onto certain signals. Ignores suit sensors, but is much less accurate."
-	icon_state = "pinpointer_syndicate"
-	worn_icon_state = "pinpointer_black"
-	minimum_range = 25
-	has_owner = TRUE
-	ignore_suit_sensor_level = TRUE
+/datum/contractor_item/comms_blackout
+	name = "Comms Outage"
+	desc = "Request Syndicate Command to disable station Telecommunications. Disables telecommunications across the station for a medium duration."
+	item_icon = "phone-slash"
+	limited = 2
+	cost = 2
 
-/obj/item/storage/box/contractor/fulton_extraction
-	name = "Fulton Extraction Kit"
-	icon_state = "syndiebox"
-	illustration = "writing_syndie"
+/datum/contractor_item/comms_blackout/handle_purchase(datum/contractor_hub/hub)
+	. = ..()
 
-/obj/item/storage/box/contractor/fulton_extraction/PopulateContents()
-	new /obj/item/extraction_pack(src)
-	new /obj/item/fulton_core(src)
+	if(!(.))
+		return
+	var/datum/round_event_control/event = locate(/datum/round_event_control/communications_blackout) in SSevents.control
+	event.runEvent()
```

Comment:
This feels wrong.

Reply:
It's how it's done in modern code from what I see